src/openrct2/actions/TileModifyAction.cpp

```diff
@@ -236,5 +238,13 @@ GameActions::Result TileModifyAction::QueryExecute(bool isExecuting) const
     res.Position.y = _loc.y;
     res.Position.z = TileElementHeight(_loc);
 
+    if (isExecuting)
+    {
+        MapInvalidateTileFull(_loc);
+        auto intent = Intent(INTENT_ACTION_TILE_MODIFY);
+        // TODO: add location data to intent.
```

Comment:
You could use the member function `PutExtra` to send along some data, in this case the X and Y tile coords as two parameters. Don't get too fancy with this, as the `Intent` class is something to be replaced.

Reply:
This was done originally in this PR, but we decided against it since it required reworking the entire intent class to do properly. I accidentally left this todo in when I pulled the intent change out of the PR, but it's gone now.